src/core/env/src/main/scala/StreamUtilities.scala

```diff
@@ -34,40 +36,33 @@ object StreamUtilities {
 
   /** Iterate through the entries of a streamed .zip file, selecting only sampleRatio of them
     *
-    * @param portableStream  Stream of zip file
+    * @param stream  Stream of zip file
     * @param zipfile         File name is only used to construct the names of the entries
     * @param sampleRatio     What fraction of files is returned from zip
     */
-  class ZipIterator(portableStream: PortableDataStream, zipfile: String, sampleRatio: Double = 1)
+  class ZipIterator(stream: InputStream, zipfile: String, sampleRatio: Double = 1, random: Random)
```

Comment:
Please put parameters with defaults at the end. It's great Scala supports mixing optional and non-optional params like this but it leads to strange corner cases.